### ISSUE TITLE
Is there a cache for files opened within API request[QUESTION]

### ISSUE THREAD

**Issue by meteoDaniel** [tiangolo/fastapi#736]
**Description**

I am running an API that opens a netcdf4(hdf5) file after a request was submitted. To avoid memory problems during multiple queries I want to cache the opened file. So that the file is loaded only once in the memory. 
The used netcdf4 file changes every 4 hours. 

Is that possible and If yes could you give me some examples?

**Additional context**

**Comment by prostomarkeloff** [comment]
What's about size of file?

**Comment by dmontagu** [comment]
Yes, it is possible, but there is nothing built into fastapi that will do this; you'll need to be responsible for implementing the caching yourself.

The right way to do this depends a lot on context you haven't shared -- where is the file coming from? What are you doing with it? How big is it? Does it matter if multiple worker processes are each caching it separately?

Probably the easiest approach to keep the file contents in memory would be to use a `functools.lru_cache` (with a low `maxsize` to prevent storing a long history). You can find lots of blog articles, examples, etc. about how to use `lru_cache`.

If `lru_cache` doesn't work for you, the right answer probably depends heavily on the answers to the questions above, potentially along with context about your backend deployment environment, etc.

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @meteoDaniel  :+1:
